### PR TITLE
Remove no-op behaviour for no-content `<label></label>` elements

### PR DESCRIPTION
**Pull request by dougbu**
- #6638

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/LabelTagHelperTest.cs [hunk 1]
```diff
@@ -101,8 +99,6 @@ public class LabelTagHelperTest
                     { modelWithText, typeof(NestedModel), () => modelWithText.NestedModel.Text, "NestedModel.Text",
                         new TagHelperOutputContent("Hello World1", "Hello World2", "Hello World2", "NestedModel_Text") },
 
-                    // Note: Tests cases below here will not work in practice due to current limitations on indexing
-                    // into ModelExpressions. Will be fixed in https://github.com/aspnet/Mvc/issues/1345.
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/LabelTagHelperTest.cs]: Drive-by fix: #1345 has been fixed and this comment was no longer valid.

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperLabelExtensionsTest.cs [hunk 2]
```diff
@@ -155,20 +278,24 @@ public void LabelHelpers_DisplayDisplayName_IfNonNull(string displayName)
             Assert.Equal("<label for=\"\">HtmlEncode[[" + displayName + "]]</label>", HtmlContentUtilities.HtmlContentToString(labelForModelResult));
         }
 
+        // Prior to aspnet/Mvc#6638 fix, helpers generated nothing with this setup.
+        // Following test mimics use of an identity expression in an editor template if invoked for a property. See
+        // also LabelHelpers_ReturnExpectedElementForModel_IfDisplayNameEmptyAndNotTopLevel().
         [Fact]
-        public void LabelHelpers_ReturnEmptyForProperty_IfDisplayNameEmpty()
+        public void LabelHelpers_ReturnExpectedElementForProperty_IfDisplayNameEmptyAndNotTopLevel()
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperLabelExtensionsTest.cs]: Test previously returned empty because the setup was incorrect. Used neither `provider` nor `modelExplorer`.

File: test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/LabelTagHelperTest.cs [hunk 3]
```diff
@@ -234,6 +230,109 @@ public class LabelTagHelperTest
...
+                    {
+                        "a label", string.Empty, "prefix", "HtmlEncode[[a label]]", $"prefix_{nameof(NestedModel.Text)}"
+                    },
+                };
+            }
+        }
+
+        // Prior to aspnet/Mvc#6638 fix, helpers generated nothing in this test when displayName was empty.
+        [Theory]
+        [MemberData(nameof(DisplayNameDataSet))]
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/LabelTagHelperTest.cs]: nit: Could totally just use InlineData

Comment (dougbu) [reply test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/LabelTagHelperTest.cs]: Nah. I want to use `string.Empty` and interpolated strings.

**Comment by dougbu** [comment]
8645ada
